### PR TITLE
support timezone when scheduling with Argo workflows

### PR DESCRIPTION
**Pull request by amerberg**
This PR adds support for specifying a timezone when scheduling Argo workflows.

The relevant Argo documentation is here: https://argoproj.github.io/argo-workflows/cron-workflows/#cronworkflow-options

### CODE REVIEW

File: metaflow/plugins/argo/argo_workflows.py
Comment (savingoyal) [inline metaflow/plugins/argo/argo_workflows.py:183]: small nit - can we have `_cron(...)` return a tuple that includes the timezone as well. that will negate the need for creating this function.

Comment (amerberg) [reply metaflow/plugins/argo/argo_workflows.py]: Done.

**Comment by amerberg** [comment]
How do we want to handle AWS Batch, which as far as I can tell, doesn't support a timezone with the cron schedule? It seems like we should at least issue a warning, perhaps even fail, if the user specifies a timezone and we can't use it with it.

**Comment by shrinandj** [comment]
Thanks a lot for adding this!

Other than addressing the review comment above, can you also add a note about the testing you did for this change? The basic tests to do here would include running a flow providing a timezone and running a flow without a timezone (current behavior).

**Comment by amerberg** [comment]
Yes, I tested by adding a schedule decorator to a simple flow. I created workflows with two different timezones, and then one with no timezone, and confirmed that they all started at the correct time.

**Comment by savingoyal** [comment]
> How do we want to handle AWS Batch, which as far as I can tell, doesn't support a timezone with the cron schedule? It seems like we should at least issue a warning, perhaps even fail, if the user specifies a timezone and we can't use it with it.

Yep - within AWS Step Functions we would want to check if the user has specified timezone and fail hard if that is the case. Can you also add a note in `@schedule` re: the format that the timezone should be specified in.

**Comment by savingoyal** [comment]
cc @valayDave re: impact on Airflow integration

**Comment by amerberg** [comment]
Added documentation for timezone format (IANA standard) and an assertion to check that a timezone isn't provided with Step Functions.

**Review by shrinandj — DISMISSED** [review]
LGTM! Thanks for your contributions!

File: metaflow/plugins/aws/step_functions/step_functions.py
Comment (savingoyal) [inline metaflow/plugins/aws/step_functions/step_functions.py:332]: rather than the assert, can you raise `StepFunctionsException` so that Metaflow can pretty print the exception?

Comment (amerberg) [reply metaflow/plugins/aws/step_functions/step_functions.py]: Done.

**Review by savingoyal — APPROVED** [review]
LGTM! (pending a review from @valayDave regarding Airflow compatibility)

**Comment by amerberg** [comment]
Based on the [Airflow docs](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/timezone.html#time-zone-aware-dags), I think the way to make this work with Airflow would be to use a timezone-aware `pendulum.datetime` for the `start_date` here: https://github.com/Netflix/metaflow/blob/f6c632022d94f11313453084264a6fb9ae16ef28/metaflow/plugins/airflow/airflow.py#L649. However, I don't have a working Airflow setup to test it.

File: metaflow/plugins/aws/step_functions/schedule_decorator.py
Comment (amerberg) [reply metaflow/plugins/aws/step_functions/schedule_decorator.py:32]: Thanks, done.

**Comment by valayDave** [comment]
@amerberg ; Thanks for scoping the Airflow change. For now we can leave airflow with out the change. I already have some pending stacked PRs like #1234 where i can incorporate necessary changes.

**Comment by amerberg** [comment]
@valayDave Should we at least add a check, analogous to the Step Functions check, to fail if the user specifies a timezone with Airflow?

**Comment by valayDave** [comment]
@amerberg : I already have something in the works with #1252 ; For the moment we can merge this PR after which I will rebase and update my stacked PRs so that we do something analogous to the Step functions check.

**Pull request merged**
